### PR TITLE
More typing for scrapy/core/downloader

### PR DESCRIPTION
**Pull request by wRAR**

### CODE REVIEW

File: scrapy/core/downloader/handlers/http11.py [hunk 1]
```diff
@@ -123,83 +127,88 @@ class TunnelingTCP4ClientEndpoint(TCP4ClientEndpoint):
...
         )
         protocol.transport.write(tunnelReq)
         self._protocolDataReceived = protocol.dataReceived
-        protocol.dataReceived = self.processProxyResponse
+        protocol.dataReceived = self.processProxyResponse  # type: ignore[method-assign]
         self._protocol = protocol
         return protocol
 
-    def processProxyResponse(self, rcvd_bytes):
+    def processProxyResponse(self, data: bytes) -> None:
```
Comment (Gallaecio) [inline scrapy/core/downloader/handlers/http11.py:158]: Technically backward incompatible, but :see_no_evil:

File: scrapy/core/downloader/handlers/http11.py [hunk 2]
```diff
@@ -230,18 +239,20 @@ class TunnelingAgent(Agent):
...
-        pool=None,
+        reactor: ReactorBase,
+        proxyConf: Tuple[str, int, Optional[bytes]],
+        contextFactory: Optional[IPolicyForHTTPS] = None,
+        connectTimeout: Optional[float] = None,
+        bindAddress: Optional[bytes] = None,
+        pool: Optional[HTTPConnectionPool] = None,
     ):
+        # TODO make this arg required instead
+        assert contextFactory is not None
```
Comment (Gallaecio) [inline scrapy/core/downloader/handlers/http11.py:250]: To do in this PR, or should we create an issue to track this?

**Review by Gallaecio — APPROVED** [review]

File: scrapy/core/downloader/handlers/http11.py (hunk 2 above)
Comment (wRAR) [reply scrapy/core/downloader/handlers/http11.py:250]: Not here, I'll make an issue for both instances. Not sure how to properly do this though. We could make all args kw-only.

Comment (wRAR) [reply scrapy/core/downloader/handlers/http11.py:250]: https://github.com/scrapy/scrapy/issues/6343

File: scrapy/core/downloader/handlers/http11.py (hunk 1 above)
Comment (wRAR) [reply scrapy/core/downloader/handlers/http11.py:158]: I assume it's only called when assigned to `protocol.dataReceived` so there shouldn't be problems even theoretically.

**Pull request merged**